pkg/cmd/operator/cleanupjob.go

```diff
@@ -118,11 +122,42 @@ func (o *CleanupJobOptions) Run(streams genericclioptions.IOStreams, cmd *cobra.
 	stopCh := signals.StopChannel()
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+
 	go func() {
 		<-stopCh
 		cancel()
 	}()
 
+	var errs []error
+
+	err := o.runCleanup(ctx)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("can't run the cleanup: %w", err))
+	}
+
+	select {
+	case <-stopCh:
```

Comment:
Might be more future proof to use `ctx.Done()` as that's the one that runCleanup uses and someone might say wrap it with a timeout in the future.

Reply:
This is intentional, there shouldn't be any cleanups running when runCleanup ends without an error. It would be ambigous if users would see that cleanups ongoing are stopped just before jobs ends. So I stop it only when stop signal comes.
Lets handle future code changes when they happen.

Reply:
anyways, should it check the error is not nil? it can happen that the stop channel gets hit after the cleanup exits successfully

Reply:
added

Reply:
could just be an else branch but this works for me, thx